src/io/flutter/run/bazel/FlutterBazelRunConfigurationType.java

```diff
@@ -32,14 +32,25 @@ public Factory(FlutterBazelRunConfigurationType type) {
     @Override
     @NotNull
     public RunConfiguration createTemplateConfiguration(@NotNull Project project) {
+      // This is always called first when loading a run config, even when it's a non-template config.
+      // See RunManagerImpl.doCreateConfiguration
       return new BazelRunConfig(project, this, "Flutter (Bazel)");
     }
 
     @Override
     @NotNull
     public RunConfiguration createConfiguration(String name, RunConfiguration template) {
-      // Override the default name which is always "Unnamed".
-      return super.createConfiguration(template.getProject().getName(), template);
+      // Called in two cases:
+      //   - When creating a non-template config from a template.
+      //   - whenever the run configuration editor is open (for creating snapshots).
+      // In the first case, we want to override the defaults from the template.
+      // In the second case, don't change anything.
+      //   (But the name doesn't matter; it will immediately be overwritten.)
+      if ((name.equals("Unnamed") || name.startsWith("Unnamed (")) && template instanceof BazelRunConfig) {
```

Comment:
Changed the hack so it should work with an internationalized name. (Not tested that way, though.)

There seems to be nothing in the run configuration framework that lets you know if you're creating a new, regular configuration from a template configuration. There are two factory methods but they don't actually work that way. Hence the comments and this hack.

It looks like LocatableConfiguration and setGeneratedName() aren't used for this codepath. As far as I can tell, they are used by automatically generated run configs (for example, right-clicking on a file or method), not when creating a new run config by clicking '+'. The name being passed in here is generated by RunConfigurable.createUniqueName() which is private.
 
(It all seems rather dubious to me, and hardly anything is commented, so it's hard to determine intent. If JetBrains wants people to do things differently they should write javadoc explaining what to do.)